interfaces/IBrowser.h

```diff
@@ -149,6 +151,26 @@ namespace Exchange {
 
         // @brief Initiate garbage collection
         virtual uint32_t CollectGarbage() = 0;
+
+        // @brief Run javascript in main frame
+        // @param script utf8 encoded JS code string
+        virtual uint32_t RunJavaScript(const string& script) = 0;
+
+        // @brief Add user script to be executed at document start
+        // @param script utf8 encoded JS code string
+        virtual uint32_t AddUserScript(const string& script, const bool topFrameOnly) = 0;
+        // @brief Remove all user scripts
+        virtual uint32_t RemoveAllUserScripts() = 0;
+
+        // @property
+        // @brief HTTP cookie jar
+        // @param version Version of the cookie jar payload
+        // @param checksum The checksum of the string used for payload creation
+        // @param payload Base64 string representation of encrypted and compressed list of cookies
+        /* @json:omit */
+        virtual uint32_t CookieJar(uint32_t& version /* @out */, int32_t& checksum /* @out */, string& payload /* @out */) const = 0;
+        /* @json:omit */
+        virtual uint32_t CookieJar(const uint32_t version, const int32_t checksum, const string& payload) = 0;
```

Comment:
Same argument as above lets avoid GOD interfaces and split. Hence why we have COMRPC QueryInterface..
Suggest to put the CookJar and its notification a a dedicated interface, like IBrowserScripting, something like ICookieJar? Looking at the methods, is this to set and get a a whole cookieJar? Not a single cookie I guess? Is that usefull on an interface?

Reply:
>  Looking at the methods, is this to set and get a a whole cookieJar? Not a single cookie I guess? Is that usefull on an interface?

yes, it is meant for storing cookie jar in the "cloud"